### PR TITLE
User should be logged in to their default store

### LINKED ISSUE #2099: Users should login to their default stores as set in mSupply Cloud?
## What went wrong? 😲
If there are multiple stores within a site with multiple users, I think the users should be able to login to their default store after signing in with their credentials. (since oms doesn't have option to choose store to login unlike the mSupply desktop with stores DDL to choose to login)

## Expected behaviour 🤔
Users should login to their default store as set in mSupply Cloud!

## How to Reproduce 🔨
Steps to reproduce the behaviour:
1. Have multiple stores within a site with multiple users, for eg: have `user1`'s default store as `store1` and `user2` on `store2`
2. Go to login page for oms
3. Throw in the credentials for `user1`
4. You'll (by default) login to `store1` 
5. Log out
6. Now throw in credentials for 'user2'
7. See error: you'll still login to 'store1' (even when default store is set as 'store2`)

## Screenshots/Recording
https://recordit.co/Bd8iQ1P7V3

## Your environment 🌱

 - Version:  v1.3.00-test
 - Platform:
   - [ ] android (tablet)
   - [x] browser (extra points if you tell us which one)
   - [x] desktop (windows)
   - [ ] desktop (macOS)
   - [x] server (windows)

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2099

# 👩🏻‍💻 What does this PR do? 
 
The issue raised is due to the most recent credentials being a single object. If you are switching users then the previous user's store was being set as the current store. This overrides the user's default store - which is only used when the user hasn't previously logged in.

The change is to store every user's store when they login, and then match the username when setting the current store. If the user does not have a recent entry, then their default store is used.

# 🧪 How has/should this change been tested? 

You'll need at least two users, with one of them having multiple stores available.
- Clear local storage: `localStorage.removeItem('@openmsupply-client/mru/credentials');`
- Login as user_a, see that they are assigned their default store
- logout, login as user_b, see that they are assigned their default store
- change the store (store_b), logout and login. You should be in store_b
- login as user_a: they'll be in user_a::default_store

etc..

Basically - the first time a user logs in, the store should be their default.
If a user changes stores, then logs out and in again, the store should be the most recently used store.

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: client/packages/common/src/authentication/api/hooks/useLogin.ts [hunk 1]
```diff
@@ -53,22 +53,31 @@ export const useLogin = (
...
 
   // returns MRU store, if set
   // or the first store in the list
   const getStore = async (userDetails?: Partial<UserNode>) => {
     const defaultStore = userDetails?.defaultStore;
     const stores = userDetails?.stores?.nodes;
+    const mru = mostRecentlyUsedCredentials?.find(
+      item =>
+        item.username.toLowerCase() === userDetails?.username?.toLowerCase()
+    );
```
Comment (CarlosNZ) [inline client/packages/common/src/authentication/api/hooks/useLogin.ts:77]: The only problem is when you switch branches without clearing local storage, this throws an error, as `.find()` is not a valid function of the previous `'/mru/credentials'` object (you've changed it to an array). This would be a problem the first time a user tries to use a version with this change in it, wouldn't it?

Perhaps just add an additional check here to ensure that `mostRecentlyUsedCredentials` is an array, and if not , set it to an empty array?

**Review by CarlosNZ — COMMENTED** [review]
Looks good, thanks. Works well. Just one comment about migration -- would be good to make it a little more seamless from the users' POV. :)

File: client/packages/common/src/authentication/api/hooks/useLogin.ts (hunk 1 above)
Comment (mark-prins) [reply client/packages/common/src/authentication/api/hooks/useLogin.ts:77]: oh.. good point! on to it

Comment (mark-prins) [reply client/packages/common/src/authentication/api/hooks/useLogin.ts]: @CarlosNZ accounted for previous object format in 5b1a1c84a
Is that ok?

**Review by CarlosNZ — APPROVED** [review]
Great, thanks for the fix.

Wonder if it might be a good idea at some stage to add some additional type checking in `useLocalStorage` to prevent future mismatches like this. Probably a bit overkill for now, but something to think about down the line if we encounter similar migration issues.

**Comment by mark-prins** [comment]
yes, fair point. tricky to do that neatly.. thanks javascript for being so flexible 🙄
